src/io/inputs/yml-system/converter.cpp

```diff
@@ -121,17 +122,106 @@ static SystemModel::Component createComponent(const YmlSystem::Component& c,
     return component;
 }
 
+/**
+ * @brief Creates a SystemModel::Connection from a YmlSystem::Connection and a map of components.
+ *
+ * This function constructs a SystemModel::Connection by looking up components and ports
+ * based on the IDs provided in the YmlSystem::Connection. It ensures that the ports are
+ * of the same type and that fields are correctly configured for sending and receiving.
+ *
+ * @param connection A YmlSystem::Connection object containing the connection details.
+ * @param components An unordered map of component IDs to SystemModel::Component objects.
+ *
+ * @return A SystemModel::Connection object representing the created connection.
+ *
+ * @throw std::invalid_argument if a component or port is not found, if the ports are not
+ *        of the same type, or if fields are incorrectly configured for sending/receiving.
+ */
+static SystemModel::Connection createConnection(
```

Comment:
About **createConnection(...)** : 
This function is long.
Indeed it does 2 things (many checks and one creation at the end), and defines lambdas.
- Lambdas ; 
  have to be short (that's why they have no name) and are usually called from algorithms. Here lambdas have names and are not particularly short.
We could make them free functions and define them out of **createConnection(...)**.
- Couldn't we make checks in different functions, and have something like : 
```c++
for (const auto& connection: ymlSystem.connections)
{
    checkComponentsExistence(connection, components);
    checkPortsExistence(connection, components);
    checkPortsCompatibility(connection, components);
    connections.push_back(createConnection(connection, components));
}
```
?

Reply:
the code is fine